File: Web App/src/models/card.ts
export type CardType =
    | 'LEGEND'
    | 'UNIT'
    | 'SPELL'
    | 'GEAR'
    | 'RUNE'
    | 'BATTLEFIELD';

export type Domain =
    | 'MIND'
    | 'CALM'
    | 'BODY'
    | 'ORDER'
    | 'CHAOS'
    | 'FURY'
    | 'COLORLESS';

export type RiftRarity =
    | 'COMMON'
    | 'UNCOMMON'
    | 'RARE'
    | 'EPIC'
    | 'SHOWCASE'
    | 'TOKEN';

export interface Card {
    cardId: string;
    name: string;
    type: CardType;
    domain: Domain[];
    cost: number;
    might: number;
    text: string;
    flavorText: string | null;
    power: number | null;
    health: number | null;
    rarity: RiftRarity;
    artistName: string | null;
    setCode: string;
    collectorNumber: string;
    imageUrl: string;
    foilImageUrl: string | null;
    isAlternativeArt: boolean;
}

export interface BinderEntry {
    binderId: string;
    cardId: string;
    quantity: number;
}

export interface CardWithQuantity {
    card: Card;
    quantity: number;
}

File: Android App/app/src/main/java/com/lburne/bounded/data/model/Card.kt
package com.lburne.bounded.data.model

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "card_definitions") // Renamed table
data class Card(
    @PrimaryKey val cardId: String, // e.g., "OGN-001"
    val name: String,
    val setCode: String,
    val collectorNumber: Int,
    val imageUrl: String, // URL or local path
    val domain: List<Domain>,
    val rarity: RiftRarity,
    val type: CardType,
    val cost: Int,
    val might: Int?,
    val artistName: String?
)